reference/configuration/security.rst

```diff
@@ -22,19 +22,376 @@ key in your application configuration.
+                    # the rest of options depend on the authentication mechanism
+                    # ...
+
+    .. code-block:: xml
+
+        <!-- app/config/security.xml -->
+        <?xml version="1.0" encoding="UTF-8"?>
+        <srv:container xmlns="http://symfony.com/schema/dic/security"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xmlns:srv="http://symfony.com/schema/dic/services"
+            xsi:schemaLocation="http://symfony.com/schema/dic/services
+                http://symfony.com/schema/dic/services/services-1.0.xsd">
+
+            <config>
+                <!-- ... -->
+
+                <!-- 'pattern' is a regular expression matched against the incoming
+                     request URL. If there's a match, authentication is triggered -->
+                <firewall name="main" pattern="^/admin">
+                    <!-- the rest of options depend on the authentication mechanism -->
+                    <!-- ... -->
+                </firewall>
+            </config>
+        </srv:container>
+
+    .. code-block:: php
+
+        // app/config/security.php
+
+        // ...
+        $container->loadFromExtension('security', [
+            'firewalls' => [
+                // 'main' is the name of the firewall (can be chosen freely)
+                'main' => [
+                    // 'pattern' is a regular expression matched against the incoming
+                    // request URL. If there's a match, authentication is triggered
+                    'pattern' => '^/admin',
+                    // the rest of options depend on the authentication mechanism
+                    // ...
+                ],
+            ],
+        ]);
+
+.. tip::
+
+    Read :doc:`this article </security/firewall_restriction>` to learn about how
+    to restrict firewalls by host and HTTP methods.
+
+In addition to some common config options, the most important firewall options
+depend on the authentication mechanism, which can be any of these:
+
+.. code-block:: yaml
+
+    # app/config/security.yml
+
+        security:
+            # ...
+            firewalls:
+                main:
+                    # ...
+                        x509:
+                            # ...
+                        remote_user:
+                            # ...
+                        simple_preauth:
+                            # ...
+                        guard:
+                            # ...
+                        form_login:
+                            # ...
+                        form_login_ldap:
+                            # ...
+                        json_login:
+                            # ...
+                        simple_form:
+                            # ...
+                        http_basic:
+                            # ...
+                        http_basic_ldap:
+                            # ...
+                        http_digest:
+                            # ...
```

Comment:
I think it's maybe better to do this oneline: `x509: # ...`. That way, we make the code block half it's current size :)

Reply:
Yes ... but we don't follow that format in the rest of the docs, so I prefer to be consistent.